File: py/src/intro/starter.py
import boto3
import pprint

bedrock = boto3.client(
    service_name='bedrock',
    region_name='eu-central-1')

pp = pprint.PrettyPrinter(depth=4)

def list_foundation_models():
    models = bedrock.list_foundation_models()
    for model in models["modelSummaries"]:
        pp.pprint(model)
        pp.pprint("-------------------")

def get_foundation_model(modelIdentifier):
    model = bedrock.get_foundation_model(modelIdentifier=modelIdentifier)
    pp.pprint(model)

# list_foundation_models()
# get_foundation_model('anthropic.claude-v2')

File: ts/src/intro/starter.ts
import { BedrockClient, ListFoundationModelsCommand, GetFoundationModelCommand } from '@aws-sdk/client-bedrock'

const client = new BedrockClient({
    region: 'eu-central-1'
});

async function listFoundationModels(){
    const response = await client.send(
        new ListFoundationModelsCommand({})
    )
    console.log(response)
}

async function getModelInfo(modelName: string){
    const response = await client.send(
        new GetFoundationModelCommand({
            modelIdentifier:modelName
        })
    )
    console.log(response)
}

getModelInfo('anthropic.claude-v2');